paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/PrepareResourcesTask.kt

```diff
@@ -42,7 +42,7 @@ open class PrepareResourcesTask : DefaultTask() {
         .use {
           it.write(project.packageName())
           it.newLine()
-          it.write(mergeResourcesProvider.get().outputDir.path)
+          it.write(mergeResourcesProvider.get().outputDir.get().asFile.path)
```

Comment:
would hate to force all users to a minimum of AGP 3.6.  will need to think about this.  @AlecStrong any learnings from how SqlDelight manages breaking API changes across AGP versions?

Reply:
@jrodbx asked me to take a look, since I've also had to deal with breaking changes in API in my own plugin, which supports AGP 3.5.3, 3.6, and the latest 4.0 beta. While I have not solved the problem in a general way,* fortunately the changes have been small enough I've been able to create little workarounds in the two cases it has come up. For example, when AGP 3.6 changed the type of the `compile<variant>JavaWithJavac` task from `AndroidJavaCompile` to `JavaCompile` (from core Gradle), that was an issue. I dealt with that [here](https://github.com/autonomousapps/dependency-analysis-android-gradle-plugin/blob/master/src/main/kotlin/com/autonomousapps/internal/dependencyAnalysis.kt#L149-L155) by grabbing the task by name alone rather than name and type, and used the generalized `Task#getOutputs()` (`TaskOutputs`) API provided by Gradle to get access to its file outputs.

In another case, both the name of a task changed, _and_ one of its output types, and I chose to use reflection to get the strongly typed property (instead of using `TaskOutputs`). You can see that [here](https://github.com/autonomousapps/dependency-analysis-android-gradle-plugin/blob/master/src/main/kotlin/com/autonomousapps/internal/dependencyAnalysis.kt#L189) and [here](https://github.com/autonomousapps/dependency-analysis-android-gradle-plugin/blob/master/src/main/kotlin/com/autonomousapps/internal/agpUtils.kt#L21).

You can get the current AGP version available at runtime by importing `com.android.builder.model.ANDROID_GRADLE_PLUGIN_VERSION`.

*I have spoken to a few people on the Gradle community slack who have solved this problem in a general way by having a complex build with many subprojects. There is an interface / adapter between the main plugin code and AGP, and the various subprojects each implement a different version of the interface, one per AGP version. The correct subproject is used at runtime. This is a completely general solution, but obviously a bit on the complex side. They did it because it was proprietary and they sold it as a product, and had customers on various versions of AGP.

Reply:
I added a reflection based solution in the last commit which should work for agp<3.5. Please take a look.